### PR TITLE
User profile loaded using GET

### PR DESCRIPTION
**Pull request by derrick-w**
Profile retrieved using GET request on profile endpoint

### CODE REVIEW

**Review by shchau — APPROVED** [review]
Looks fine to me

File: frontend/src/pages/Profile.jsx [hunk 1]
```diff
@@ -61,26 +69,48 @@ class Profile extends Component {
...
+                </div>
+                </div>
             </Container>
 	    )
     }
 }
 
-export default Profile;
+const mapStateToProps = (state) => {
+    return {
```
Comment (htruong4) [inline frontend/src/pages/Profile.jsx:111]: I don't think I see this props value (this.props.userId from mapStateToProps) being used anywhere. Are you putting this here in the case you need it for the future?

File: frontend/src/pages/Profile.jsx [hunk 2]
```diff
@@ -61,26 +69,48 @@ class Profile extends Component {
...
+        this.getProfile();
+    }
+
+    getProfile() {
+        var urlPath = "/api/author/"
+        var authorId = store.getState().loginReducers.userId.split("thor/");
+        const path = urlPath + authorId[1], requireAuth = true;
+        HTTPFetchUtil.getRequest(path, requireAuth)
+        .then((httpResponse) => {
+            if (httpResponse.status === 200) {
```
Comment (htruong4) [inline frontend/src/pages/Profile.jsx:82]: I think you should have an else block here to catch for the case of the status being 400 or something not 200.

File: frontend/src/pages/Profile.jsx [hunk 3]
```diff
@@ -61,26 +69,48 @@ class Profile extends Component {
...
+                httpResponse.json().then((results) => {
+                    this.setState( {
+                        profiledata: results
+                    })
+                })
```
Comment (htruong4) [inline frontend/src/pages/Profile.jsx:87]: In the case that the promise returned from .json() gets rejected for whatever reason, it's good practice to have a .catch block to handle the rejected promises.

**Review by htruong4 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Profile.jsx [hunk 4]
```diff
@@ -42,15 +48,17 @@ class Profile extends Component {
                         <Table.Body>
                         <Table.Row>
                             <Table.Cell>Name</Table.Cell>
-                            <Table.Cell>Henry Truong</Table.Cell>
+                            <Table.Cell>{this.state.profiledata.firstName} {this.state.profiledata.lastName}</Table.Cell>
                         </Table.Row>
                         <Table.Row>
                             <Table.Cell>Github</Table.Cell>
-                            <Table.Cell><a href="https://github.com/htruong1">https://github.com/htruong1</a></Table.Cell>
+                            <Table.Cell><a href={this.state.profiledata.github}>{this.state.profiledata.github}</a>
```
Comment (hiufungk) [inline frontend/src/pages/Profile.jsx:55]: maybe open the link in a new tab instead?

File: frontend/src/pages/Profile.jsx (hunk 2 above)
Comment (htruong4) [reply frontend/src/pages/Profile.jsx:82]: It also might be good to maybe build or use  a component to display an error message in the case we get a 400 or something.

File: frontend/src/pages/Profile.jsx (hunk 4 above)
Comment (derrick-w) [reply frontend/src/pages/Profile.jsx:55]: kk

File: frontend/src/pages/Profile.jsx (hunk 1 above)
Comment (derrick-w) [reply frontend/src/pages/Profile.jsx:111]: left it in b/c I thought it was necessary for grabbing user id from store, removed

**Review by forgeno — COMMENTED** [review]
I have the same change suggestions as @htruong1

File: frontend/src/pages/Profile.jsx (hunk 2 above)
Comment (derrick-w) [reply frontend/src/pages/Profile.jsx:82]: your wish is my command

File: frontend/src/pages/Profile.jsx (hunk 3 above)
Comment (derrick-w) [reply frontend/src/pages/Profile.jsx:87]: this is our fight

**Pull request closed without merging**